<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1029</width>
    <height>953</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>230</y>
      <width>381</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>360</y>
      <width>381</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>490</y>
      <width>381</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>0</y>
      <width>381</width>
      <height>231</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="picker">
     <item row="0" column="0" colspan="3">
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Выбор алгоритма</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" colspan="3">
      <widget class="QComboBox" name="algoPick">
       <item>
        <property name="text">
         <string>Параметрическое уравнение</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Каноническое уравнение</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Алгоритм средней точки</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Алгоритм Брезенхема</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Библиотечная функция</string>
        </property>
       </item>
      </widget>
     </item>
     <item row="2" column="0" colspan="3">
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Выбор цвета</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>R</string>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>G</string>
       </property>
      </widget>
     </item>
     <item row="3" column="2">
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>B</string>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QSpinBox" name="redPick">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QSpinBox" name="greenPick">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
     </item>
     <item row="4" column="2">
      <widget class="QSpinBox" name="bluePick">
       <property name="maximum">
        <number>255</number>
       </property>
       <property name="value">
        <number>255</number>
       </property>
      </widget>
     </item>
     <item row="6" column="0" colspan="3">
      <widget class="ColorWidget" name="colorView" native="true"/>
     </item>
     <item row="5" column="0">
      <widget class="QPushButton" name="backgroundColor">
       <property name="text">
        <string>Фон</string>
       </property>
      </widget>
     </item>
     <item row="5" column="1">
      <widget class="QPushButton" name="blueColor">
       <property name="text">
        <string>Синий</string>
       </property>
      </widget>
     </item>
     <item row="5" column="2">
      <widget class="QPushButton" name="blackColor">
       <property name="text">
        <string>Черный</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>251</y>
      <width>381</width>
      <height>106</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="circle">
     <item row="0" column="0" colspan="3">
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Окружность</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>X0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Y0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>R</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLineEdit" name="circleX0">
       <property name="text">
        <string>200</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="circleY0">
       <property name="text">
        <string>200</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLineEdit" name="circleR">
       <property name="text">
        <string>30</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="3">
      <widget class="QPushButton" name="circleDraw">
       <property name="text">
        <string>Построить</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>380</y>
      <width>381</width>
      <height>106</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="ellipse">
     <item row="0" column="0" colspan="4">
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Эллипс</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_11">
       <property name="text">
        <string>X0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_12">
       <property name="text">
        <string>Y0</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_13">
       <property name="text">
        <string>A</string>
       </property>
      </widget>
     </item>
     <item row="1" column="3">
      <widget class="QLabel" name="label_14">
       <property name="text">
        <string>B</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLineEdit" name="ellipseX0">
       <property name="text">
        <string>300</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="ellipseY0">
       <property name="text">
        <string>300</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLineEdit" name="ellipseA">
       <property name="text">
        <string>100</string>
       </property>
      </widget>
     </item>
     <item row="2" column="3">
      <widget class="QLineEdit" name="ellipseB">
       <property name="text">
        <string>50</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="4">
      <widget class="QPushButton" name="ellipseDraw">
       <property name="text">
        <string>Построить</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>520</y>
      <width>381</width>
      <height>108</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="circleGroup">
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="label_15">
       <property name="text">
        <string>Спектр концентрических окружностей</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_17">
       <property name="text">
        <string>RN</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="circleGroupStep">
       <property name="text">
        <string>10</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_18">
       <property name="text">
        <string>Step</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLineEdit" name="circleGroupRN">
       <property name="text">
        <string>300</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="2">
      <widget class="QPushButton" name="circleGroupDraw">
       <property name="text">
        <string>Построить</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>670</y>
      <width>381</width>
      <height>108</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="ellipseGroup">
     <item row="2" column="0">
      <widget class="QLineEdit" name="ellipseGroupAStep">
       <property name="text">
        <string>10</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_20">
       <property name="text">
        <string>Шаг А</string>
       </property>
      </widget>
     </item>
     <item row="2" column="2">
      <widget class="QLineEdit" name="ellipseGroupN">
       <property name="text">
        <string>10</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_21">
       <property name="text">
        <string>Шаг В</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLineEdit" name="ellipseGroupBStep">
       <property name="text">
        <string>10</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="3">
      <widget class="QLabel" name="label_19">
       <property name="text">
        <string>Cкептр концентрических эллиспов</string>
       </property>
      </widget>
     </item>
     <item row="1" column="2">
      <widget class="QLabel" name="label_22">
       <property name="text">
        <string>N</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" colspan="3">
      <widget class="QPushButton" name="ellipseGroupDraw">
       <property name="text">
        <string>Построить</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>640</y>
      <width>381</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>790</y>
      <width>381</width>
      <height>59</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="time">
     <item row="0" column="0">
      <widget class="QLabel" name="label_16">
       <property name="font">
        <font>
         <weight>75</weight>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Измерение временных характеристик</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QPushButton" name="timeDraw">
       <property name="text">
        <string>Измерить</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="clearButton">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>860</y>
      <width>381</width>
      <height>41</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Очистить</string>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
    <property name="default">
     <bool>false</bool>
    </property>
    <property name="flat">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QStackedWidget" name="show_widget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>611</width>
      <height>881</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="page">
     <widget class="PixelDrawer" name="drawer" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>611</width>
        <height>881</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="page_2">
     <widget class="ProfileWidget" name="profiler" native="true">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>611</width>
        <height>881</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1029</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ColorWidget</class>
   <extends>QWidget</extends>
   <header>colorwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>PixelDrawer</class>
   <extends>QWidget</extends>
   <header>pixeldrawer.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ProfileWidget</class>
   <extends>QWidget</extends>
   <header>profilewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
